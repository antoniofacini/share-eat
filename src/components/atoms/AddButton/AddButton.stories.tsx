import { Story, Meta } from '@storybook/react/types-6-0'
import AddButton from '.'

export default {
  title: 'Atoms/AddButton',
  component: AddButton,
} as Meta

export const Default: Story = (args) => <AddButton {...args} />
Default.args = { isVariant: true }
