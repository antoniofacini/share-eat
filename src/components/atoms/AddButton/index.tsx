interface IAddButton {
  isVariant?: boolean
  handler: () => void
}

const AddButton = ({ isVariant, handler }: IAddButton) => {
  return isVariant ? (
    <div className="w-fit p-4 border-black rounded-full bg-yellow" onClick={handler}>
      <svg
        aria-label="adicionar mais items"
        width="25"
        height="25"
        viewBox="0 0 20 20"
        fill="none"
        xmlns="https://www.w3.org/2000/svg"
      >
        <path
          d="M11 2C11 1.44772 10.5523 1 10 1C9.44772 1 9 1.44772 9 2H11ZM9 18C9 18.5523 9.44772 19 10 19C10.5523 19 11 18.5523 11 18H9ZM18 11C18.5523 11 19 10.5523 19 10C19 9.44772 18.5523 9 18 9V11ZM2 9C1.44772 9 1 9.44772 1 10C1 10.5523 1.44772 11 2 11V9ZM9 2V10H11V2H9ZM9 10V18H11V10H9ZM10 11H18V9H10V11ZM10 9H2V11H10V9Z"
          fill="#121212"
        />
      </svg>
    </div>
  ) : (
    <div className="w-fit p-2 border-white border-2 rounded-full" onClick={handler}>
      <svg
        aria-label="adicionar mais items"
        width="16"
        height="16"
        viewBox="0 0 20 20"
        fill="none"
        xmlns="https://www.w3.org/2000/svg"
      >
        <path
          d="M11 2C11 1.44772 10.5523 1 10 1C9.44772 1 9 1.44772 9 2H11ZM9 18C9 18.5523 9.44772 19 10 19C10.5523 19 11 18.5523 11 18H9ZM18 11C18.5523 11 19 10.5523 19 10C19 9.44772 18.5523 9 18 9V11ZM2 9C1.44772 9 1 9.44772 1 10C1 10.5523 1.44772 11 2 11V9ZM9 2V10H11V2H9ZM9 10V18H11V10H9ZM10 11H18V9H10V11ZM10 9H2V11H10V9Z"
          fill="#FFFFFF"
        />
      </svg>
    </div>
  )
}

export default AddButton
