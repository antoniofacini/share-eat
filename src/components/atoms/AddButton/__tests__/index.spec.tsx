import { render } from '@testing-library/react'
import AddButton from '..'

describe('<AddButton />', () => {
  it('should render AddButton component', () => {
    const wrapper = render(<AddButton handler={() => console.log('handle')} />)
    expect(wrapper).toMatchSnapshot()
  })
})
