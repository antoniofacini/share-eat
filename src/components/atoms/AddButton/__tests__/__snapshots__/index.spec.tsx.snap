// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddButton /> should render AddButton component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-fit p-2 border-white border-2 rounded-full"
      >
        <svg
          aria-label="adicionar mais items"
          fill="none"
          height="16"
          viewBox="0 0 20 20"
          width="16"
          xmlns="https://www.w3.org/2000/svg"
        >
          <path
            d="M11 2C11 1.44772 10.5523 1 10 1C9.44772 1 9 1.44772 9 2H11ZM9 18C9 18.5523 9.44772 19 10 19C10.5523 19 11 18.5523 11 18H9ZM18 11C18.5523 11 19 10.5523 19 10C19 9.44772 18.5523 9 18 9V11ZM2 9C1.44772 9 1 9.44772 1 10C1 10.5523 1.44772 11 2 11V9ZM9 2V10H11V2H9ZM9 10V18H11V10H9ZM10 11H18V9H10V11ZM10 9H2V11H10V9Z"
            fill="#FFFFFF"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-fit p-2 border-white border-2 rounded-full"
    >
      <svg
        aria-label="adicionar mais items"
        fill="none"
        height="16"
        viewBox="0 0 20 20"
        width="16"
        xmlns="https://www.w3.org/2000/svg"
      >
        <path
          d="M11 2C11 1.44772 10.5523 1 10 1C9.44772 1 9 1.44772 9 2H11ZM9 18C9 18.5523 9.44772 19 10 19C10.5523 19 11 18.5523 11 18H9ZM18 11C18.5523 11 19 10.5523 19 10C19 9.44772 18.5523 9 18 9V11ZM2 9C1.44772 9 1 9.44772 1 10C1 10.5523 1.44772 11 2 11V9ZM9 2V10H11V2H9ZM9 10V18H11V10H9ZM10 11H18V9H10V11ZM10 9H2V11H10V9Z"
          fill="#FFFFFF"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
